### ISSUE TITLE
Remove txt_matches(search_type, &format!("fn {}", searchstr), signature)

### ISSUE THREAD

**Issue by kngwyu** [racer-rust/racer#941]
It's inefficient and incorrect.

**Comment by kngwyu** [comment]
e.g.
It's used here:

**Comment by h-michael** [comment]
@kngwyu 
Hi thinks maintain racer.

I also would like to contribute from now on, but can this issue simply remove this method?

**Comment by kngwyu** [comment]
No, I mean it doesn't match `fn   methodname`or so, and it should return match position.
If you're intested in contributing, I also recommend #956.

**Comment by h-michael** [comment]
@kngwyu 
I see, I'll try it!
